lib/tree.rb

```diff
@@ -1,38 +1,63 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age = @age + 1
+    @height = @height + 10
+    if age > 2
+      add_apples!
+    end
+    if age > 5
+      @alive = false
+    end
   end
 
-  def add_apples
+  def add_apples!
+    i = 0
+    until i > 6 + Random.rand(4)
```

Comment:
Managing the looping yourself like this isn't typical in Ruby. Here are a couple other options:

```ruby
0.upto(6 + rand(4)) do
end
```
```ruby
(0..(6 + rand(4)).each do
end
```